### PR TITLE
[SourceKit] Stop printing normal comments in clang generated interface

### PR DESCRIPTION
**Pull request by rintaro**
Generated interfaces for Clang modules used to try printing normal
comments between decls extracted from the header text. That was because
doc-comment was not common in C/ObjC headers. But mainly because of
"import as member feature" Clang decls aren't printed in the order as
they appear in the header file, the logic determinig which comment
belongs to which decl was not working property. We've decided to remove
that feature and only print the proper doc-comments as it has been
getting common.

rdar://93731287

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/SourceKit/InterfaceGen/gen_clang_module.swift.response [hunk 1]
```diff
@@ -112,18 +96,10 @@ public var fooIntVar: Int32
...
-/*
-  Very good
-  fooFuncWithBlock function.
-*/
 public func fooFuncWithBlock(_ blk: ((Float) -> Int32)!)
-
 public func fooFuncWithFunctionPointer(_ fptr: (@convention(c) (Float) -> Int32)!)
-
 public func fooFuncNoreturn1() -> Never
 public func fooFuncNoreturn2() -> Never
```
Comment (rintaro) [inline test/SourceKit/InterfaceGen/gen_clang_module.swift.response:104]: No blank line between top-level decls doesn't seem to be expected (i.e. swift modules show blank lines). Looking into it

**Comment by rintaro** [comment]
https://github.com/apple/llvm-project/pull/8443
@swift-ci Please smoke test

File: utils/run-test [hunk 2]
```diff
@@ -258,7 +258,7 @@ def main():
     if args.verbose:
         test_args = ["-a"]
     else:
-        test_args = ["-sv"]
+        test_args = ["-svv"]
```
Comment (ahoppen) [inline utils/run-test:261]: Unrelated change?

**Review by ahoppen — APPROVED** [review]

File: test/SourceKit/InterfaceGen/gen_clang_module.swift.response (hunk 1 above)
Comment (rintaro) [reply test/SourceKit/InterfaceGen/gen_clang_module.swift.response:104]: There are some difference between Swift decl printing and clang decl printing in `swift::ide::printModuleInterface()`, Swift decls always have trailing newline, but clang decls don't. Fixing it is out of scope of this PR.

**Comment by rintaro** [comment]
https://github.com/apple/llvm-project/pull/8443
@swift-ci Please smoke test

**Review by hamishknight — APPROVED** [review]

**Comment by rintaro** [comment]
https://github.com/apple/llvm-project/pull/8443
@swift-ci Please smoke test Linux
